index.html

```diff
@@ -0,0 +1,475 @@
+
+### `csdl_SelectionSet`
+
+```graphql
+#::[`scalar csdl_SelectionSet` declaration]
+scalar csdl_SelectionSet
+  @specifiedBy(url: "https://specs.apollo.dev/csdl/v1#csdl_selectionset")
+```
+
+A GraphQL selection set represented with the same syntax as defined in [the GraphQL SDL](https://spec.graphql.org/draft/#sec-Selection-Sets).
+
+Note: Unlike the federation `_FieldSet` scalar, a `csdl_SelectionSet` must parse as a GraphQL selection set, so the selections must be surrounded by braces.
+
+All legal selection set features are allowed. That includes alises, field arguments, directives, and inline fragments. Named fragment spreads are syntactically valid, but meaningless, as there is nowhere to define named fragments; future versions of this spec may change this.
+
+### `csdl_URL`
+
+```graphql
+#::[`scalar csdl_URL` declaration]
+scalar csdl_URL
+  @specifiedBy(url: "https://specs.apollo.dev/csdl/v1#csdl_url")
+```
+
+A [URL](https://www.w3.org/Addressing/URL/url-spec.html). Used by `@csdl_endpoint`.
+
+## Directives
+
+Valid CSDL documents **must** include the declarations of any CSDL directives they use.
+
+### `@csdl_key`
+
+```graphql
+#::[`@csdl_key` declaration]
+directive @csdl_key(graph: csdl_Graph!, repr: csdl_SelectionSet!)
+  repeatable on OBJECT
+```
+
+Define an entity key for this type within a subgraph.
+
+The `@csdl_key` directive tells consumers what subset of fields are necessary to identify this type of entity to a particular subgraph. It provides a way for csdl consumers to "switch graphs" when planning a query. For example:
+
+```graphql
+#::[A query which requires porting the `Astronaut` type between subgraphs]
+query {          # resolver                    | realizes
+                 # ----------------------------|--------------
+  rockets {      # rockets::Query.rockets     -> rockets::Rocket
+    captain {    # rockets::Rocket.astronaut  -> rockets::Astronaut
+      name       # astronauts::Astronaut.name -> String
+      rocket {   # rockets::Astronaut.rocket  -> rockets::Rocket
+        id       # rockets::Rocket.id         -> ID
+      }
+    }
+  }
+  
+}
+```
+
+`Astronaut.name` is provided by the `astronauts` subgraph. But the `captain` field is provided by the `rockets` subgraph.
+
+The fields specified in `repr` will be passed to the subgraph's `Query._entities(representations:)` as an item within the `representations` list.
+
+Multiple `@csdl_key`s can be provided for different graphs, or for the same graph.
+
+```graphql
+#::[Using `@csdl_key` to specify subgraph keys]
+type Astronaut
+  @csdl_key(graph: ASTRONAUTS, repr: "{ id }")
+  @csdl_key(graph: ROCKETS, repr: "{ id }")
+{ ... }
+```
+
+### `@csdl_endpoint`
+
+```graphql
+#::[`@csdl_endpoint` declaration]
+directive @csdl_endpoint(url: csdl_URL) on ENUM_VALUE
+```
+
+Bind an endpoint URL to a subgraph. This directive is only valid on enum values within the required `csdl_Graph` enum type.
```

Comment:
"is only valid" — does that mean that consumers should reject any CSDL applying this directive elsewhere?